docs/architecture/incremental.md

```diff
@@ -0,0 +1,23 @@
+# Configuration: Streams & Incremental Syncs
+
+## AirbyteStream
+* The `AirbyteStream` has two sections:
+    * The `discovered` section describes information and metadata about the data in the source. This section is generated by the source connector. Outside of the source, this section should be treated as read only.
+    * The `configuration` section describes functionality that an end user would like applied on the sync. This section is configured by the end-user, and is never touched by the source.
+
+## Discovered
+
+Currently this section contains two fields: `json_schema` and `metadata`.
+* `json_schema` describes the structure of the data in a source.
+* `metadata` describes other information about the data in the source. For example what features are supported for the stream. Currently the `metadata` object contains two fields.
```

Comment:
`metadata` is one of my top 5 least favorite words in the english language. it effectively means "data that i couldn't be bothered to name". encourage suggestion of other names.

Reply:
`sync_parameters` or `sync_configuration` is probably a more apt description, unless you truly intend for this to be a generic bag of data which happens to have only these two parameters right now

Reply:
Do they even have to be grouped? Can they be top level?

Reply:
moot with new structure.